cmd/admin-handlers.go

```diff
@@ -244,32 +245,86 @@ func (a adminAPIHandlers) ServerInfoHandler(w http.ResponseWriter, r *http.Reque
 	if objectAPI == nil {
 		return
 	}
+	var jsonBytes []byte
```

Comment:
Older `mc` will break with this change. To be backward compatible, we should provide `server` type information if the `infoType` query parameter is not supplied.

Reply:
Added a new api for service endpoints which accepts query parameter , as `etcd-info` ,`lambda-info`  and more will be added. The old code is kept same which fetches `server info ` using ServerInfoHandler.